op_crates/fetch/21_blob.js

```diff
@@ -0,0 +1,289 @@
+        );
+        bytesArrays.push(new Uint8Array(buffer));
+      } else if (element instanceof Blob) {
+        bytesArrays.push(
+          new Uint8Array(element[_byteSequence].buffer.slice(0)),
+        );
+      } else if (typeof element == "string") {
+        let s = element;
+        if (endings == "native") {
+          s = convertLineEndingsToNative(s);
+        }
+        bytesArrays.push(utf8Encoder.encode(s));
+      } else {
+        throw new TypeError("Unreachable code (invalild element type)");
+      }
+    }
+    return concatUint8Arrays(...bytesArrays);
+  }
+
+  /**
+   * @param {string} str 
+   * @returns {string}
+   */
+  function normalizeType(str) {
+    let normalizedType = str;
+    if (!/^[\x20-\x7E]*$/.test(str)) {
+      normalizedType = "";
+    }
+    return normalizedType.toLowerCase();
+  }
+
+  const _byteSequence = Symbol("[[ByteSequence]]");
+
+  class Blob {
+    /** @type {string} */
+    #type;
+
+    /** 
+     * @param {BlobPart[]} [blobParts]
+     * @param {BlobPropertyBag} [options]
+     */
+    constructor(blobParts, options) {
+      if (blobParts === undefined) {
+        blobParts = [];
+      }
+      if (typeof blobParts !== "object") {
+        throw new TypeError(
+          `Failed to construct 'Blob'. blobParts cannot be converted to a sequence.`,
+        );
+      }
+
+      const parts = [];
+      const iterator = blobParts[Symbol.iterator]?.();
+      if (iterator === undefined) {
+        throw new TypeError(
+          "Failed to construct 'Blob'. The provided value cannot be converted to a sequence",
+        );
+      }
+      while (true) {
+        const { value: element, done } = iterator.next();
+        if (done) break;
+        if (
+          ArrayBuffer.isView(element) || element instanceof ArrayBuffer ||
+          element instanceof Blob
+        ) {
+          parts.push(element);
+        } else {
+          parts.push(String(element));
+        }
+      }
+
+      if (!options || typeof options == "function") {
+        options = {};
+      }
+      if (typeof options != "object") {
+        throw new TypeError(
+          `Failed to construct 'Blob'. options is not an object.`,
+        );
+      }
+      const endings = options?.endings?.toString() ?? "transparent";
+      const type = options?.type?.toString() ?? "";
+
+      /** @type {Uint8Array} */
+      const bytes = processBlobParts(parts, endings);
+      this[_byteSequence] = bytes;
+      this.#type = normalizeType(type);
+    }
+
+    /** @returns {number} */
+    get size() {
+      return this[_byteSequence].byteLength;
+    }
+
+    /** @returns {string} */
+    get type() {
+      return this.#type;
+    }
+
+    /** 
+     * @param {number} [start]
+     * @param {number} [end]
+     * @param {string} [contentType]
+     * @returns {Blob}
+     */
+    slice(start, end, contentType) {
+      const O = this;
+      /** @type {number} */
+      let relativeStart;
+      if (start == undefined) {
```

Comment:
```suggestion
      if (start === undefined) {
```

Unless you are trying to trap `null` as well, but in that case, more idomatic Deno would be:

```suggestion
      if (start == null) {
```